### PR TITLE
New tests

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### Related issues
#47

### CODE REVIEW

File: tests/hikari/impl/test_rest.py [hunk 1]
```diff
@@ -770,12 +852,16 @@ async def test__request_when__token_is_None(self, rest_client, exit_exception):
         rest_client.buckets.is_started = True
         rest_client._token = None
         rest_client._acquire_client_session = mock.Mock(return_value=mock_session)
-        with mock.patch.object(asyncio, "gather", new=mock.AsyncMock()):
-            with pytest.raises(exit_exception):
-                await rest_client._request(route)
+        rest_client._stringify_http_message = mock.Mock()
+        with mock.patch.object(rest, "_LOGGER", new=mock.Mock(getEffectiveLevel=mock.Mock(return_value=100))) as logger:
```
Comment (nekokatt) [inline tests/hikari/impl/test_rest.py:856]: exit stack?

File: tests/hikari/impl/test_rest.py [hunk 2]
```diff
@@ -817,8 +911,12 @@ class StubResponse:
...
         rest_client._parse_ratelimits = mock.AsyncMock()
         rest_client._acquire_client_session = mock.Mock(return_value=mock_session)
-        with mock.patch.object(asyncio, "gather", new=mock.AsyncMock()):
-            assert (await rest_client._request(route)) is None
+        rest_client._stringify_http_message = mock.Mock()
+        with mock.patch.object(rest, "_LOGGER", new=mock.Mock(getEffectiveLevel=mock.Mock(return_value=100))) as logger:
+            with mock.patch.object(asyncio, "gather", new=mock.AsyncMock()):
+                assert (await rest_client._request(route)) is None
+
+        assert logger.log.call_count == 0
```
Comment (nekokatt) [inline tests/hikari/impl/test_rest.py:919]: assert_not_called?

File: tests/hikari/impl/test_rest.py [hunk 3]
```diff
@@ -872,9 +978,13 @@ class StubResponse:
...
-        with mock.patch.object(asyncio, "gather", new=mock.AsyncMock()):
-            with pytest.raises(exit_exception):
-                await rest_client._request(route)
+        rest_client._stringify_http_message = mock.Mock()
+        with mock.patch.object(rest, "_LOGGER", new=mock.Mock(getEffectiveLevel=mock.Mock(return_value=100))) as logger:
+            with mock.patch.object(asyncio, "gather", new=mock.AsyncMock()):
+                with pytest.raises(exit_exception):
+                    await rest_client._request(route)
+
+        assert logger.log.call_count == 0
```
Comment (nekokatt) [inline tests/hikari/impl/test_rest.py:987]: shouldn't you be testing the response, not logging?

Logging may change without notice, or be reworded. Ideally we shouldn't be testing that as the main test case, but I would assume you want to check the response of `rest_client._request` to ensure that it returns the right stuff in this case, right?

File: tests/hikari/test_channels.py [hunk 4]
```diff
@@ -185,6 +185,16 @@ async def test_TextChannel_history():
     )
 
 
+def test_TextChannel_trigger_typing():
+    app = mock.Mock()
+    app.rest.trigger_typing = mock.Mock()
+    mock_channel = mock.Mock(channels.TextChannel, id=123, app=app)
+
+    channels.TextChannel.trigger_typing(mock_channel)
```
Comment (nekokatt) [inline tests/hikari/test_channels.py:193]: mock_class_namespace could have been used here :)

**Comment by nekokatt** [comment]
This kind of collides in one or two places with what i am doing, and what i am doing resolves one or two of these issues so I have re-targeted my branch for this PR if you don't mind :)

feel free to continue adding stuff to said branch. You'll see what I've changed.

File: tests/hikari/test_channels.py (hunk 4 above)
Comment (davfsa) [reply tests/hikari/test_channels.py:193]: Ik, I would have used it, but its to keep consistency with the rest of the tests there (which idk who made them). Im probably going to end up rewriting them to look nicer

File: tests/hikari/impl/test_rest.py (hunk 3 above)
Comment (davfsa) [reply tests/hikari/impl/test_rest.py:987]: This test does that, the logging is just random testing for an extra case. I have no idea why i didnt make a separate test for it (which im going to do now)

File: tests/hikari/impl/test_rest.py (hunk 2 above)
Comment (FasterSpeeding) [reply tests/hikari/impl/test_rest.py:919]: btw just a random point but if you're testing logging there are other places where that's being done in the repo already encase you want to change how it's being tested in either place for consistency https://github.com/nekokatt/hikari/blob/69a9ac65ddb1a79022508ab592415f336f23d00d/tests/hikari/utilities/test_event_stream.py#L203

Comment (davfsa) [reply tests/hikari/impl/test_rest.py:919]: This doesnt specifically test logging, its to follow some paths on the code. Im going to change this to its own test

**Comment by nekokatt** [comment]
Ping me when this isn't a draft and I'll merge it.

**Comment by davfsa** [comment]
Idk how nor when I closed this, but was a mistake :sweat_smile: 

@nekokatt ready for review

File: tests/hikari/impl/test_rest.py [hunk 5]
```diff
@@ -129,6 +142,18 @@ def test_http_settings_property(self, rest_provider, rest_client):
...
         assert rest_provider.proxy_settings == rest_client.proxy_settings
 
+    def test_entity_factory_property(self, rest_provider, entity_factory):
+        assert rest_provider.entity_factory == entity_factory
+
+    def test_cache_property(self, rest_provider, cache):
+        assert rest_provider.cache == cache
+
+    def test_executor_property(self, rest_provider, executor):
+        assert rest_provider.executor == executor
```
Comment (nekokatt) [inline tests/hikari/impl/test_rest.py:152]: probs should use `is` here.

**Review by nekokatt — APPROVED** [review]

**Pull request merged**